android/src/app/organicmaps/car/screens/search/SearchOnMapScreen.java

```diff
@@ -0,0 +1,189 @@
+package app.organicmaps.car.screens.search;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.car.app.CarContext;
+import androidx.car.app.constraints.ConstraintManager;
+import androidx.car.app.model.Action;
+import androidx.car.app.model.CarIcon;
+import androidx.car.app.model.CarLocation;
+import androidx.car.app.model.Header;
+import androidx.car.app.model.ItemList;
+import androidx.car.app.model.Metadata;
+import androidx.car.app.model.Place;
+import androidx.car.app.model.PlaceMarker;
+import androidx.car.app.model.Row;
+import androidx.car.app.model.Template;
+import androidx.car.app.navigation.model.PlaceListNavigationTemplate;
+import androidx.core.graphics.drawable.IconCompat;
+import androidx.lifecycle.LifecycleOwner;
+
+import app.organicmaps.R;
+import app.organicmaps.bookmarks.data.MapObject;
+import app.organicmaps.car.SurfaceRenderer;
+import app.organicmaps.car.screens.base.BaseMapScreen;
+import app.organicmaps.car.util.UiHelpers;
+import app.organicmaps.location.LocationHelper;
+import app.organicmaps.search.NativeSearchListener;
+import app.organicmaps.search.SearchEngine;
+import app.organicmaps.search.SearchRecents;
+import app.organicmaps.search.SearchResult;
+
+public class SearchOnMapScreen extends BaseMapScreen implements NativeSearchListener
+{
+  private final int MAX_RESULTS_SIZE;
+
+  @NonNull
+  private final String mQuery;
+  private final boolean mIsCategory;
+
+  @Nullable
+  private ItemList mResults = null;
+
+  private SearchOnMapScreen(@NonNull Builder builder)
+  {
+    super(builder.mCarContext, builder.mSurfaceRenderer);
+    final ConstraintManager constraintManager = getCarContext().getCarService(ConstraintManager.class);
+    MAX_RESULTS_SIZE = constraintManager.getContentLimit(ConstraintManager.CONTENT_LIMIT_TYPE_PLACE_LIST);
+
+    mQuery = builder.mQuery;
+    mIsCategory = builder.mIsCategory;
+  }
+
+  @NonNull
+  @Override
+  public Template onGetTemplate()
+  {
+    final PlaceListNavigationTemplate.Builder builder = new PlaceListNavigationTemplate.Builder();
+    builder.setHeader(createHeader());
+    builder.setMapActionStrip(UiHelpers.createMapActionStrip(getCarContext(), getSurfaceRenderer()));
+    if (mResults == null)
+      builder.setLoading(true);
+    else
+      builder.setItemList(mResults);
+    return builder.build();
+  }
+
+  @Override
+  public void onResultsUpdate(@NonNull SearchResult[] results, long timestamp)
+  {
+    if (mResults != null)
+      return;
```

Comment:
What's the logic here?

Reply:
We can't update the screen each time new results come.
We're only allowed to do it once.
We also can't cancel the search as search result points on the map will be removed.

I'll prepare a separate issue for this screen with a better description of the problem and possible solutions for it.

Reply:
Wow, it's a big issue.

Reply:
We may need to change the search batching logic.